index.js

```diff
@@ -105,31 +120,75 @@ function initApp(callback) {
     /*
      * configure routes
      */
-    app.post('/api/v1/substitution', (req, res) => {
-      if (!req.isAuthenticated() || req.user.level < config.user.level.substitute) {
-        res.status(401).send('{"error":"not authenticated or not allowed"}');
-        return;
-      }
+     var Compendium = require('./lib/model/compendium');
 
-      // TODO implement ...
-    });
-    
-    app.get('/api/v1/compedium/:id/substition', function (req, res) {
+    app.post('/api/v1/substitution', (req, res) => {
       res.setHeader('Content-Type', 'application/json');
-      
-      // query substitions for given id from database
-
-      // create response
-      var response = {};
-      res.send(response);
+      // if (!req.isAuthenticated()) {
+      //   res.status(401).send('{"error":"not authenticated"}');
+      //   return;
+      // }
+      // if (req.user.level < config.user.level.substitute) {
+      //   res.status(401).send('{"error":"not allowed"}');
+      //   return;
+      // }
+
+      // new random id
+      let newID = randomstring.generate(config.id_length);
+
+      // TODO: check if id exists
+      let passon = {
+        user: "0000-0003-2287-369X",  // TODO: req.user.id
```

Comment:
:thinking: Mhhm... please change asap! Also, for testing, I recommend that you run bouncer locally and actually log in instead of using such a hack. If you want to I can show you how to set that up.

Reply:
done